### PR TITLE
added rpe-service-auth-provider dependency

### PR DESCRIPTION
**Pull request by akdasari**
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: ccd/requirements.yaml [hunk 1]
```diff
@@ -9,4 +9,7 @@ dependencies:
     repository: '@hmctspublic'
     tags:
       - ccd-idam-pr
+  - name: rpe-service-auth-provider
+    version: ~0.1.1
+    repository: '@hmctspublic'
```
Comment (markry-kainos) [inline ccd/requirements.yaml]: make dependency optional (with tags) so consumers can ignore (e.g. where they define their own s2s dep)

Comment (akdasari) [reply ccd/requirements.yaml]: modified as optional ,please review

File: ccd/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: A Helm chart for the HMCTS CCD product
 name: ccd
-version: 3.0.4
+version: 3.0.5
```
Comment (markry-kainos) [inline ccd/Chart.yaml]: pls use 3.1.0 - minor updates will get updated automatically

File: ccd/values.yaml [hunk 3]
```diff
@@ -1,6 +1,8 @@
 tags:
   postgresql-pod: true
   ccd-idam-pr: false
+  rpe-service-auth-provider: false
```
Comment (markry-kainos) [inline ccd/values.yaml]: like it - i didnt actually know this would work

**Pull request merged**